frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -914,11 +914,7 @@ const InitializationResourceAlert: React.FC<InitializationResourceAlertProps> =
         <CreateInitializationResourceButton
           obj={props.csv}
           initializationResource={initializationResource}
-          targetNamespace={
-            model?.namespaced
-              ? initializationResource?.metadata.namespace || csv.metadata?.namespace
-              : null
-          }
+          targetNamespace={csv.metadata?.namespace || initializationResource?.metadata.namespace}
```

Comment:
This looks like it will do the wrong thing for cluster-scoped resources. You also haven't updated the `canCreateCustomResource` check above.

Is the problem that `model` is undefined? We could check `!model || model.namespaced` instead or wait for the model to load. Presumably the link won't work until the model is loaded anyway.

cc @TheRealJon

Reply:
> Is the problem that `model` is undefined?
In the create MCH example the `targetNamespace` was pointing to "recommended namespace" `/k8s/ns/open-cluster-management/...` incorrectly

In the create MCE example the `model` was undefined so `targetNamespace` was `null`

Reply:
Also, with this logic, we would never use the namespace provided in the initialization resource. This might be an issue because that initialization resource is an annotation provided by the operator, and it might be using a specific namespace for a reason, meaning that overriding it transparently like this could break the operator.

Reply:
For the MCE example, the model has `namespaced: false` which I assume is why it default to putting `all-namespaces` in the link

Reply:
Oh, I see. It's going to the operand create page instead of the standard create page, so you'd want to keep the namespace in the URL even if it's not a namespaced resource.

Reply:
Ah, so we don't even need to consider the initialization resource namespace at all, right?  It would just be the same namespace as the current CSV.

Reply:
@TheRealJon made updates and pushed. PTAL